internal/runners/auth/auth.go

```diff
@@ -91,6 +94,14 @@ func (a *Auth) authenticate(params *AuthParams) error {
 		return auth.AuthenticateWithToken(params.Token, a.Auth)
 	}
 
+	if apiKey := os.Getenv(constants.APIKeyEnvVarName); apiKey != "" {
+		err := auth.AuthenticateWithToken(apiKey, a.Auth)
+		if err != nil {
+			return locale.WrapError(err, "err_auth_api_key", "Failed to authenticate with [ACTIONABLE]{{.V0}}[/RESET] environment variable", constants.APIKeyEnvVarName)
+		}
+		return nil
+	}
```

Comment:
I wonder if this error reporting should be in the lower level `auth` package. The user would only get this error message if they explicitly ran `state auth` but we are attempting to authenticate with the api key environment variable through multiple code paths in the `auth` package. See [here](https://github.com/ActiveState/cli/blob/master/pkg/platform/authentication/auth.go#L218) for one.

Reply:
That's what I did originally, but I couldn't trigger it without `state auth`. We have lots of places that check `auth.Authenticated()` and bail out with "you must be logged in" if not. I was only able to show my message using `state auth`. The only callers of your linked method are "Sync" (not applicable to this PR), and "Client". All instances of the latter appear to be walled behind `auth.Authenticated()` checks, so in practice, I think we're okay.

Given that the confusion arises around the whole `state auth` flow, I figured this was an okay place to put it.